File: testdata/fixtures/python/vulnerable/hardcoded_secret.py
# Vulnerable: Hardcoded secrets in Python source code
# Expected: GTSS-SEC-001 (HardcodedPassword), GTSS-SEC-005 (JWTSecret)

import jwt

SECRET_KEY = "my_super_secret_key_do_not_share_12345"

API_KEY = "sk_live_4eC39HqLyjWDarjtT1zdp7dc"

DATABASE_PASSWORD = "Pr0duct10n_DB_P@ssw0rd!"


class Config:
    secret_key = "flask_secret_a1b2c3d4e5f6g7h8i9j0"
    jwt_secret = "jwt_signing_key_xyz789"
    api_secret = "api_s3cret_k3y_f0r_pr0d"


def get_db_connection():
    password = "db_admin_password_2024!"
    return f"postgresql://admin:{password}@db.production.internal:5432/myapp"


def create_jwt(user_id):
    payload = {"user_id": user_id, "exp": 3600}
    return jwt.encode(payload, "hardcoded_jwt_secret_key_abc123", algorithm="HS256")


def decode_jwt(token):
    return jwt.decode(token, "hardcoded_jwt_secret_key_abc123", algorithms=["HS256"])

File: batou-rules/testdata/fixtures/ruby/vulnerable/hardcoded_secret.rb
# Vulnerable: Hardcoded secrets in source code
# Expected: GTSS-SEC-001 (HardcodedPassword), GTSS-SEC-005 (JWTSecret)

class Application < Rails::Application
  config.secret_key_base = "a1b2c3d4e5f6a7b8c9d0e1f2a3b4c5d6e7f8a9b0c1d2e3f4a5b6c7d8e9f0"

  config.api_key = "sk_live_R4nd0mStr1ngTh4tL00ksR34l1st1c"
end

class JwtService
  secret_key = "my_super_secret_jwt_key_2024_production"

  def self.encode(payload)
    JWT.sign(payload, "hardcoded_jwt_secret_key_abc123")
  end

  def self.decode(token)
    JWT.decode(token, "hardcoded_jwt_secret_key_abc123")
  end
end

module DatabaseConfig
  password = "Pr0duct10n_DB_P@ssw0rd!"
  DATABASE_URL = "postgres://admin:#{password}@db.internal:5432/myapp"
end
